lib/next_ls/signature_help.ex

```diff
@@ -0,0 +1,64 @@
+defmodule NextLS.SignatureHelp do
+  @moduledoc false
+
+  alias GenLSP.Enumerations.MarkupKind
+  alias GenLSP.Structures.MarkupContent
+  alias GenLSP.Structures.ParameterInformation
+  alias GenLSP.Structures.SignatureHelp
+  alias GenLSP.Structures.SignatureInformation
+  alias NextLS.ASTHelpers
+
+  def fetch_mod_and_name(uri, position) do
+    with {:ok, text} <- File.read(URI.parse(uri).path),
```

Comment:
you'll want to pass in the document from the handle_request callback (should be something like `lsp.assigns.documents[uri] |> Enum.join("\n")`.

and in the tests, use a didOpen and then didChange notifcation to get the server state ready to send the signature help request.